### ISSUE TITLE
Calculation of loss_linear

### ISSUE THREAD

**Issue by applefl** [ByungKwanLee/Causal-Unsupervised-Segmentation#1]
The calculation of loss_linear needs to get the label. But the unsupervised training process doesn't have label. How can I train the model with no label data?

**Comment by ByungKwanLee** [comment]
```python            
# linear probe loss
linear_logits = segment.linear(seg_feat_ema)
linear_logits = F.interpolate(linear_logits, label.shape[-2:], mode='bilinear', align_corners=False)
flat_linear_logits = linear_logits.permute(0, 2, 3, 1).reshape(-1, args.n_classes)
flat_label = label.reshape(-1)
flat_label_mask = (flat_label >= 0) & (flat_label < args.n_classes)
loss_linear = F.cross_entropy(flat_linear_logits[flat_label_mask], flat_label[flat_label_mask])
```

This is just for `Linear Probe Loss` to validate competitive dense representation quality learned
in unsupervised manners as described in Table 2 of [Paper](https://arxiv.org/pdf/2310.07379v1.pdf).

If you do not want to compute linear probe loss, then you can easily remove it.

**Comment by applefl** [comment]
Thank you. I tried to removed it, but the trainning process of fine_tuning_tr got an error at line 83, "No inf checks were recorded for this optimizer."

**Comment by ByungKwanLee** [comment]
It would be better to run the provided bash file without the change, since the Loss of Linear Probe has a different purpose (validating dense representative quality).

**Comment by xouyang0079** [comment]
Hi,

I carefully checked the STEGO and HP implementation, and they all detached the code (i.e., the output of the segmentation head) from the computation graph that lead through the "net". Hence, any operation done on the tensor after the detachment can no longer influence the "net" gradients. However, I didn't see the seg_feat_ema was detached in your implementation. Please clarify me if I am wrong.

Thanks

**Comment by ByungKwanLee** [comment]
You can find the following line in `train_front_door_tr.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/train_front_door_tr.py#L210C1-L213C67)]
```python
# Bank and EMA
cluster.bank_init()
ema_init(segment.head, segment.head_ema)
ema_init(segment.projection_head, segment.projection_head_ema
```

Here, the funcion of `ema_init` is defined in `modules/segment_module.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/modules/segment_module.py#L342)]

```python
def ema_init(x, x_ema):
    for param, param_ema in zip(x.parameters(), x_ema.parameters()): param_ema.data = param.data; param_ema.requires_grad = False
```

**Comment by xouyang0079** [comment]
> You can find the following line in `train_front_door_tr.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/train_front_door_tr.py#L210C1-L213C67)]
> 
> ```python
> # Bank and EMA
> cluster.bank_init()
> ema_init(segment.head, segment.head_ema)
> ema_init(segment.projection_head, segment.projection_head_ema
> ```
> 
> Here, the funcion of `ema_init` is defined in `modules/segment_module.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/modules/segment_module.py#L342)]
> 
> ```python
> def ema_init(x, x_ema):
>     for param, param_ema in zip(x.parameters(), x_ema.parameters()): param_ema.data = param.data; param_ema.requires_grad = False
> ```

Thanks very much for your clarification!
